src/leds.c

```diff
@@ -308,6 +308,62 @@ static void anim_felony(Leds *leds, const LedStrip *strip, const LedBar *bar, fl
     }
 }
 
+static const uint32_t rainbow_cycle_colors[] = {
+    0x00FFFF00,
+    0x0000FF00,
+    0x0000FFFF,
+    0x000000FF,
+    0x00FF00FF,
+    0x00FF0000,
+};
+static const uint8_t rainbow_cycle_colors_len =
+    sizeof(rainbow_cycle_colors) / sizeof(rainbow_cycle_colors[0]);
+
+static void anim_rainbow_cycle(Leds *leds, const LedStrip *strip, float time) {
+    uint8_t color_idx = (uint8_t) (time / 0.1f) % rainbow_cycle_colors_len;
+    strip_set_color(leds, strip, rainbow_cycle_colors[color_idx], strip->brightness, 1.0f);
+}
+
+static const uint32_t rainbow_fade_colors[] = {
+    0x00FF5000,  // FLAME
+    0x00FF7800,  // ORANGE
+    0x00FFA000,  // YELLOW
+    0x0000FF00,  // GREEN
+    0x0000FFFF,  // CYAN
+    0x008000FF,  // VIOLET
+    0x00FF70A0,  // LAVENDER
+};
+static const uint8_t rainbow_fade_colors_len =
+    sizeof(rainbow_fade_colors) / sizeof(rainbow_fade_colors[0]);
+
+static void anim_rainbow_fade(Leds *leds, const LedStrip *strip, float time) {
+    float phase = fmodf(time, (float) rainbow_fade_colors_len);
+    uint8_t curr_color_idx = (uint8_t) floorf(phase);
+    uint8_t next_color_idx = (curr_color_idx + 1) % rainbow_fade_colors_len;
+
+    // extract rbg components from current color
+    uint32_t curr_color = rainbow_fade_colors[curr_color_idx];
+    uint8_t r_curr = (curr_color >> 16) & 0xFF;
+    uint8_t g_curr = (curr_color >> 8) & 0xFF;
+    uint8_t b_curr = curr_color & 0xFF;
+
+    // extract rbg components from next color
+    uint32_t next_color = rainbow_fade_colors[next_color_idx];
+    uint8_t r_next = (next_color >> 16) & 0xFF;
+    uint8_t g_next = (next_color >> 8) & 0xFF;
+    uint8_t b_next = next_color & 0xFF;
+
+    // smoothly interpolate between the current and next color
+    float ratio = phase - curr_color_idx;
+    float smooth_ratio = (1.0f - cosf(ratio * M_PI)) / 2.0f;
+    uint8_t r = (uint8_t) ((1.0f - smooth_ratio) * r_curr + smooth_ratio * r_next);
+    uint8_t g = (uint8_t) ((1.0f - smooth_ratio) * g_curr + smooth_ratio * g_next);
+    uint8_t b = (uint8_t) ((1.0f - smooth_ratio) * b_curr + smooth_ratio * b_next);
+    uint32_t color = (r << 16) | (g << 8) | b;
+
+    strip_set_color(leds, strip, color, strip->brightness, 1.0f);
```

Comment:
Check out the `color_wheel()` function in `leds.c`, it could use a better documentation, but what it does is it gives you the whole rainbow in an indexed range of `0-255`, should be exactly what you need.

Reply:
Oh sweet - let me check that out...

Reply:
Thanks for that @lukash, was able to remove loads of code, and it still looks pretty good.
Was more RGB than "rainbow" though, so I renamed it to "RGB Fade".